core/blockchain/impl/block_tree_impl.cpp

```diff
@@ -407,4 +476,38 @@ namespace kagome::blockchain {
     }
   }
 
+  std::vector<primitives::BlockHash> BlockTreeImpl::getLeavesSorted() const {
+    std::vector<std::pair<primitives::BlockNumber, primitives::BlockHash>>
+        leaf_depths;
+    auto leaves = getLeaves();
+    leaf_depths.reserve(leaves.size());
+    for (auto &leaf : leaves) {
+      auto leaf_node = tree_->getByHash(leaf);
+      leaf_depths.emplace_back(leaf_node->depth_, leaf);
```

Comment:
Depth is a public member, right?

```suggestion
      leaf_depths.emplace_back(leaf_node->depth, leaf);
```